db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/IndexRowGenerator.java

```diff
@@ -150,17 +185,23 @@ public ExecIndexRow getNullIndexRow(ColumnDescriptorList columnList,
     		RowLocation rowLocation)
     throws StandardException
     {
-		int[] baseColumnPositions = id.baseColumnPositions();
 		ExecIndexRow indexRow = getIndexRowTemplate();
-
-		for (int i = 0; i < baseColumnPositions.length; i++)
-		{
-			DataTypeDescriptor dtd =
-				columnList.elementAt(baseColumnPositions[i] - 1).getType();
-			indexRow.setColumn(i + 1, dtd.getNull());
+		DataTypeDescriptor[] columnTypes = getIndexColumnTypes();
+
+		if (columnTypes.length == 0) {
+			int[] baseColumnPositions = id.baseColumnPositions();
+			for (int i = 0; i < baseColumnPositions.length; i++) {
+				DataTypeDescriptor dtd =
+						columnList.elementAt(baseColumnPositions[i] - 1).getType();
+				indexRow.setColumn(i + 1, dtd.getNull());
+			}
+			indexRow.setColumn(baseColumnPositions.length + 1, rowLocation);
+		} else {
+			for (int i = 0; i < columnTypes.length; i++) {
```

Comment:
it's not immediately clear why you don't use id.baseColumnPositions(); in this else branch, can you maybe add a comment in the code?

Reply:
Done.